src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -68,7 +69,7 @@ boot_compinfo_init(spdid_t spdid, captblcap_t *ct, pgtblcap_t *pt, u32_t vaddr)
 	assert(*pt);
 
 	new_comp_cap_info[spdid].compinfo = &new_compinfo[spdid];
-	cos_compinfo_init(new_comp_cap_info[spdid].compinfo, *pt, *ct, 0, (vaddr_t)vaddr, 4, &boot_info);
+	cos_compinfo_init(new_comp_cap_info[spdid].compinfo, *pt, *ct, 0, (vaddr_t)vaddr, BOOT_CAPTBL_FREE, &boot_info);
```

Comment:
This is slightly inefficient in the case where we have a component with only sinv capabilities.  Now we're starting to allocate them at a higher place in the captable than is necessary.  I'm not sure how bad this is, but to figure out, we'll just need to do the math on how many capabilities are in the first captbl node from BOOT_CAPTBL_FREE to the end of the first node in the capability table.

Reply:
I would assume that we'd want two ways to create components (i.e. parameterize this function by which starting capability image it uses.

Reply:
once we determine where the set of non resource managers ends in the cap table should probably add another value to (COMP_CAPTBL_FREE?) the enum in cos_types so were not passing magic numbers like 4.